### PR TITLE
Minor optimizations for AOT method size

### PR DESCRIPTION
**Pull request by electrum**

### CODE REVIEW

File: aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java [hunk 1]
```diff
@@ -462,14 +464,23 @@ private Class<?> compileClass(FunctionSection functions) {
 
         classWriter.visitEnd();
 
-        return loadClass(className, classWriter.toByteArray());
+        try {
+            return loadClass(className, classWriter.toByteArray());
+        } catch (MethodTooLargeException e) {
```
Comment (andreaTP) [inline aot/src/main/java/com/dylibso/chicory/aot/AotMachine.java:469]: :+1:

**Review by andreaTP — APPROVED** [review]
LGTM thanks!

To review this kind of PRs I do need to squint my eyes.
It would be great to be able to compare examples of the generated bytecode.
Ideally, we can have [approval tests](https://github.com/approvals/ApprovalTests.Java), on a few selected demo modules.

**Comment by electrum** [comment]
I rebased this on #426 and added additional approval tests, which makes the bytecode changes easy to see.

**Review by andreaTP — APPROVED** [review]
LGTM!
As a general advice, approval tests are nice but tend to become high maintenance (e.g. generating larger diffs on PRs).
My rule of thumb is to stay with the minimum amount to show a proper diff when something changes but don't try to track each and every detail.
I'm ok with this if it's the first "corpus", but, please, don't abuse them :-)

**Comment by electrum** [comment]
Agreed! We have similar types of tests in Trino, and used to have a lot more for query plans, but deleted them due to the maintenance problem. I think we'll need to play with this to determine the right balance.

After rebasing this PR and seeing the diffs to the approval tests, it was more interesting and helpful than I expected, so I'm glad that you added this.

**Pull request merged**